src/item_contents_ui.cpp

```diff
@@ -0,0 +1,325 @@
+            wprintz( w_data, c_white, filter_string );
+            wprintz( w_data, c_light_gray, " >" );
+        }
+    }
+}
+
+void item_contents_ui::draw_header()
+{
+    trim_and_print( w_head, point( 1, 1 ), ui_width, c_white, pContaining_item->display_name() );
+}
+
+void item_contents_ui::draw_item_list()
+{
+    max_item_name_width = data_width - 2;
+    int item_min = 0;
+    max_items = vFilteredItemList.size();
+    int istart = 0;
+    int iend = 0;
+    if( max_items > data_lines ) {
+        if( selected_item_line <= item_min + data_lines /2 ) {
+            istart = item_min;
+            iend = item_min + data_lines;
+        } else if( selected_item_line >= max_items - data_lines / 2 ) {
+            istart = max_items - data_lines;
+            iend = max_items;
+        } else {
+            istart = selected_item_line - data_lines / 2;
+            iend = selected_item_line + data_lines / 2;
+        }
+    } else {
+        istart = 0;
+        iend = std::min<int>( vFilteredItemList.size(), data_height + 1 );
+    }
+    for( int i = istart; i < iend; ++i ) {
+        std::string tmp_name = vFilteredItemList[i]->display_name();
+        const bool highlight = i == selected_item_line;
+        const nc_color col = highlight ? h_white : c_white;
+        const point print_from( 2, i - istart );
+        if( highlight ) {
+            cursor_pos = print_from;
+            trim_and_print( w_data, print_from, max_item_name_width, col, remove_color_tags( tmp_name ) );
+        } else {
+            trim_and_print( w_data, print_from, max_item_name_width, col, tmp_name );
+        }
+    }
+}
+
+void item_contents_ui::execute()
+{
+    item_info_scroll = 0;
+
+    head_height = 3;
+    ui_width = 0;
+    data_lines = 0;
+    data_height = 0;
+    item_info_width = 0;
+
+    keybinding_x = 1;
+    ui.on_screen_resize( [&]( ui_adaptor & ui ) {
+
+        ui_width = TERMX * 4 / 5;
+        const int wStart = ( TERMX - ui_width ) / 2;
+
+        // Keybinding tips
+        static const translation inline_fmt = to_translation(
+                //~ %1$s: action description text before key,
+                //~ %2$s: key description,
+                //~ %3$s: action description text after key.
+                "keybinding", "%1$s[<color_yellow>%2$s</color>]%3$s" );
+        static const translation separate_fmt = to_translation(
+                //~ %1$s: key description,
+                //~ %2$s: action description.
+                "keybinding", "[<color_yellow>%1$s</color>]%2$s" );
+        std::vector<std::string> act_descs;
+        const auto add_action_desc = [&]( const std::string & act, const std::string & txt ) {
+            act_descs.emplace_back( ctxt.get_desc( act, txt, input_context::allow_all_keys,
+                                                   inline_fmt, separate_fmt ) );
+        };
+        add_action_desc( "CONFIRM", to_translation( "Confirm your selection" ).translated() );
```

Comment:
```suggestion
        add_action_desc( "CONFIRM", _( "Confirm your selection" ) );
```
Less code and faster.

Reply:
Done.  I hadn't tracked down exactly what the _() did previously.